Adding POX Docker Image

RicGobs opened the pull request:
Hi,

Here it is a **Docker POX image**, to allow using POX without importing POX as a Git submodule. The new image is based on the kathara/base image.

The Docker image is needed for the sdn-openflow labs (located in the main labs directory).

RicGobs committed:
add POX docker image

tcaiazzi committed:
Fix POX Dockerfile

tcaiazzi committed:
Add POX build

Skazza94 commented on pox/Dockerfile:
```diff
@@ -0,0 +1,26 @@
+ARG DEBIAN_FRONTEND="noninteractive"
+RUN apt clean all
+RUN apt update
+RUN apt install -y zlib1g-dev libncurses5-dev libgdbm-dev libnss3-dev libssl-dev libsqlite3-dev libreadline-dev libffi-dev curl libbz2-dev
```
After compiling Python3.9, I think you can remove these libraries.

Skazza94 commented on pox/Dockerfile:
```diff
@@ -0,0 +1,26 @@
+
+# Clone and set permissions for pox
+RUN git clone https://github.com/noxrepo/pox.git && \
+    chmod +x /pox/pox.py
```
Please add the following snippet at the end of the Dockerfile to clean up caches/other:
```
RUN apt clean && \
    rm -rf /tmp/* /var/lib/apt/lists/* /var/tmp/*
```

Skazza94 reviewed (changes requested):
Hi @RicGobs,
thanks for the PR. I have some changes to ask.

Thank you!
Mariano.

RicGobs committed:
add clean-cache and remove-libraries

RicGobs commented:
Hi @Skazza94,

I have done the following:
- added the snippet code at the end (for the cache)
- removed zlib1g-dev libncurses5-dev libgdbm-dev libnss3-dev libssl-dev libsqlite3-dev libreadline-dev libffi-dev curl libbz2-dev (after their use)
- done a test (it works on my labs)

Tell me if other is needed. Bye
Riccardo

Skazza94 reviewed (approved):
Hi @RicGobs,
thanks for the changes. The PR looks good! 😄
